ios-xcframework/build.sh

```diff
@@ -18,13 +18,130 @@ function _xcodebuild {
     fi
 }
 
+function log {
+  if [[ $# -eq 1 ]]; then
+    PREFIX=''
+    MESSAGE=$1
+  else
+    PREFIX=":$1: "
+    MESSAGE=$2
+  fi
+
+  set +u
+  if [[ -n "$BUILDKITE" ]]; then
+    echo "--- $PREFIX$MESSAGE"
+  else
+    echo "$MESSAGE"
+  fi
+  set -u
+}
+
 DERIVED_DATA_PATH=./DerivedData
 DESTINATION='generic/platform=iOS'
 
 rm -rf $DERIVED_DATA_PATH
 
+MAIN_FRAMEWORK_NAME=Gutenberg
+WORKSPACE="./XCFrameworkScaffold.xcworkspace"
+SCHEME=$MAIN_FRAMEWORK_NAME
+
+log "Building $SCHEME..."
+
 _xcodebuild clean build \
-  -workspace ./XCFrameworkScaffold.xcworkspace \
-  -scheme 'Gutenberg' \
+  -workspace $WORKSPACE \
+  -scheme $SCHEME \
   -destination "$DESTINATION" \
   -derivedDataPath "$DERIVED_DATA_PATH"
+
+# See https://hackernoon.com/cocoapod-as-xcframework-with-dependencies
+# and https://github.com/traveloka/ios-rn-prebuilt
+
+ARCHIVES_ROOT=archives
+IOS_DEVICE_ARCHIVE_PATH="$ARCHIVES_ROOT/ios_devices.xcarchive"
+IOS_SIMULATOR_ARCHIVE_PATH="$ARCHIVES_ROOT/ios_simulators.xcarchive"
+
+FINAL_OUTPUT="$MAIN_FRAMEWORK_NAME.xcframework"
+
+rm -rf "$ARCHIVES_ROOT"
+rm -rf "$FINAL_OUTPUT"
+
+# 1. Archive for iOS
+log "Archiving $SCHEME for iOS"
+
+_xcodebuild archive \
+  -workspace "$WORKSPACE" \
+  -scheme "$SCHEME" \
+  -configuration Release \
+  -sdk iphoneos \
+  -archivePath "$IOS_DEVICE_ARCHIVE_PATH" \
+  -derivedDataPath "$DERIVED_DATA_PATH" \
+  BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+  SKIP_INSTALL=NO
+
+# 2. Archive for Simulator
+log "Archiving $SCHEME for Simulator"
+
+_xcodebuild archive \
+  -workspace "$WORKSPACE" \
+  -scheme "$SCHEME" \
+  -configuration Debug \
+  -sdk iphonesimulator \
+  -archivePath "$IOS_SIMULATOR_ARCHIVE_PATH" \
+  -derivedDataPath "$DERIVED_DATA_PATH" \
+  BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+  SKIP_INSTALL=NO
```

Comment:
I think we could merge the above two xcodebuild commands into a for loop? And the variables `IOS_DEVICE_ARCHIVE_PATH` & `IOS_SIMULATOR_ARCHIVE_PATH` becomes `ARCHIVES_ROOT/<platfom-name>`

Reply:
I decided to use a function instead of a loop. I find bash loops awkward. 

See 6c3e43af. What do you think?